packages/outsystems-wrapper/src/legacyCordova.ts

```diff
@@ -0,0 +1,238 @@
+          directory
+        }
+
+        // @ts-ignore
+        CapacitorUtils.Synapse.Filesystem.deleteFile(success, error, options)
+    }
+
+    private getOptionalDirectoryTypeFrom(isInternal: boolean | undefined, isTemporary: boolean | undefined): Directory | undefined {
+        // Handle the case where both parameters are undefined
+        if (isInternal === undefined || isTemporary === undefined) {
+            return undefined
+        } else {
+            return this.getDirectoryTypeFrom(!!isInternal, !!isTemporary)
+        }
+    }
+
+    private getDirectoryTypeFrom(isInternal: boolean, isTemporary: boolean): Directory {
+        // @ts-ignore
+        if (cordova.platformId == 'android') {
+            if (isInternal) {
+                return isTemporary ? Directory.Cache : Directory.Data
+            }
+            return isTemporary ? Directory.ExternalCache : Directory.External
+        }
+        return isTemporary ? Directory.Temporary : Directory.LibraryNoCloud;
+    }
+
+    private readFile(success: (res: string | Blob) => void, error: (err: PluginError) => void, path: string, isInternal: boolean | undefined, isTemporary: boolean | undefined): void {
+        let directory: Directory | undefined = this.getOptionalDirectoryTypeFrom(isInternal, isTemporary);
+        let options: ReadFileInChunksOptions = {
+            path: path,
+            directory: directory,
+            chunkSize: 256 * 1024
+        }
+
+        let chunks: string[] = []        
+        let synapseSuccess = (res: ReadFileResult) => {
+            if (res.data === "") {
+                success(chunks.join(''))
+            } else if (typeof res.data === 'string') {
+                chunks.push(res.data)
+            } else {
+                chunks.push(res.data.toString())
+            }
+        }
+
+        // @ts-ignore
+        CapacitorUtils.Synapse.Filesystem.readFileInChunks(synapseSuccess, error, options)
+    }
+
+    private dataToBlobUrl(data: string | Blob, mimeType: string): string {
+        let blob: Blob
+        if (data instanceof Blob) {
+            // If the input is already a Blob, simply create a Blob URL
+            blob = data
+        } else {
+            // Decode the Base64 data to binary
+            let binaryString = atob(data) // Decodes the Base64 string            
+            let binaryArray = new Uint8Array(
+                Array.from(binaryString, char => char.charCodeAt(0))
+            );
+        
+            // Create a Blob object from the binary data
+            blob = new Blob([binaryArray], { type: mimeType })
+        }
+
+        // Generate and return the Blob URL
+        return URL.createObjectURL(blob)
+    }
+
+    private getMimeType(fromName: string) {
+        const mimeTypes: {[id:string]:string} = {
+            'txt': 'text/plain',
+            'html': 'text/html',
+            'htm': 'text/html',
+            'css': 'text/css',
+            'js': 'application/javascript',
```

Comment:
Minor I think [`application/javascript` is obsolete](https://www.iana.org/assignments/media-types/application/javascript), should change?

```suggestion
            'js': 'text/javascript',
```